### DISCUSSION TITLE
Multitenancy - Create Child tenant in Tenant

### DISCUSSION THREAD

**Discussion by TechAspirant** [serenity-is/Serenity#5733] [Help]
How can I create child tenant in Tenant ? I am referring example provided in below link
https://github.com/volkanceylan/MultiTenancy

**Comment by dfaruque** [comment]
You might add another field in IMultiTenantRow. e.g ChildTenantIdField
Or you might create another interface for IMultiTenantWithChildRow with ChildTenantIdField.
Then in your all multi-tenant row implement that.

**Reply by TechAspirant** [reply]
Creating another field will restrict to only one child whereas their might be the case that One tenant include multiple child tenant.
What about the permissions how it will behave ? All permissions from parent tenant should be available to child tenant.

**Reply by TechAspirant** [reply]
Can we implement the treegrid view in the same ?

**Reply by edwardch** [reply]
Do you want many levels or just two levels?

If many levels then look at some hierarchical table structures, a self referencing table is probably easiest to implement.

E.g
**Table**

Tenants

**Columns** 

TenantId
Name
ParentTenantId (FK on self)

**Reply by TechAspirant** [reply]
I want many levels. Do I need to modify anything in permission table ? 
How permission will work in case of many levels ?

**Reply by edwardch** [reply]
You shouldn't need to modify permissions.

With multi tenancy you typically assign a tenant id to almost all your tables. Then when a user who belongs to a tenant logs in, all the data should be filtered to only show data from their tenant.

For this you have to pass the tenant id with every request to ensure they can only see/modify their own tenant data.

If you want multi levels, you must decide if a user can see all data for their own tenant, AND all child tenants, or some other configuration.

If you want them to see data from many tenants then you need those tenant ids to be part of each of those requests somehow.
